### ISSUE TITLE
Wrong logLevel parsing if "alert" is within a log label

### ISSUE THREAD

**Issue by DylanGuedes** [grafana/grafana#45015]
**What happened**:
tl;dr: when parsing logs, if a label has any mention to "alert" on its name, Grafana sets logLevel=critical even if there's a label "level=info" in the logs.

**What you expected to happen**:
I expected it to set `logLevel` with the same value as `level`. This works if there's no "alert" mentioned in other fields.

**How to reproduce it (as minimally and precisely as possible)**:
1. Send to a loki instance a log with log line `{\"anything\":\"lol\",\"level\":\"INFO\"}` and any pair of labels (ex: `{job="newone"}`)
2. In Grafana, with Loki configured as datasource, query the pair of labels of the log line sent in the previous step: `{job="newone"}`
3. See that Grafana will parse it as logLevel=info.
4. Send to that same loki instance a log with log line `{\"anything\":\"alert\",\"level\":\"INFO\"} and any pair of labels (ex: `{job="newone"}`)
5. In Grafana, with Loki configured as datasource, query the pair of labels of the log line sent in the previous step: `{job="newone"}`
6. Check that Grafana will parse the new log line as logLevel=critical.

**Anything else we need to know?**:

**Environment**:
- Grafana version: Grafana v8.2.0 (d7f71e9eae)
- Data source type & version: Loki/logs, v2.4.2
- OS Grafana is installed on: Linux Ubuntu
- User OS & Browser: Linux Ubuntu, Chrome
- Grafana plugins: Loki
- Others:

**Comment by DylanGuedes** [comment]
> @DylanGuedes we try to extract log level from the `level` log label. If you don't have `level` label, we are trying to find out if its content matches any of the supported expressions and we go with the first match, which is in your case `alert`. What you can do is to use JSON parser that would turn level from the log content to level label and we would then be able to correctly set INFO level. e.g. `{job="newone"} | json` Can you try if that works for you?

Yes, using `| json` definitely works (I forgot to say this in the PR description, my bad).

I concluded it was a bug because it parses the logLevel for `{\"anything\":\"anything\",\"level\":\"INFO\"}` correctly but not for `{\"anything\":\"alert\",\"level\":\"INFO\"}`, regardless of having `\"level\":\"INFO\"`, just because of the word "alert".

**Comment by ivanahuckova** [comment]
Yes, that is (in your case unfortunately) correct behaviour. We try to find the matching word and if the alert is first, we will consider the log line to be critical level. If info was first, the log would have info level. But as you said, using `| json` works and that's the way to correctly get the level from your log content. 

We have also open feature request for [Custom log levels](https://github.com/grafana/grafana/issues/25585), but in your case this is no problem because you are already using the name `level` in your log line.
